cql/src/test/java/org/apache/cassandra/stargate/metrics/ClientMetricsTest.java

```diff
@@ -0,0 +1,486 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.cassandra.stargate.metrics;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.DistributionSummary;
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.search.MeterNotFoundException;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import io.stargate.db.ClientInfo;
+import io.stargate.db.metrics.api.ClientInfoMetricsTagProvider;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.cassandra.stargate.transport.internal.Server;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class ClientMetricsTest {
+
+  ClientMetrics clientMetrics = ClientMetrics.instance;
+
+  MeterRegistry meterRegistry = new SimpleMeterRegistry();
+
+  // manual mocks, init before all
+  ClientInfoMetricsTagProvider clientTagProvider;
+  Server server1;
+  Server server2;
+  ClientInfo clientInfo1;
+  ClientInfo clientInfo2;
+
+  @BeforeAll
+  public void initMocks() {
+    clientTagProvider = mock(ClientInfoMetricsTagProvider.class);
+    server1 = mock(Server.class);
+    server2 = mock(Server.class);
+    clientInfo1 = mock(ClientInfo.class);
+    clientInfo2 = mock(ClientInfo.class);
+
+    when(clientTagProvider.getClientInfoTags(clientInfo1)).thenReturn(Tags.of("client", "one"));
+    when(clientTagProvider.getClientInfoTags(clientInfo2)).thenReturn(Tags.of("client", "two"));
+
+    List<Server> servers = Arrays.asList(server1, server2);
+    clientMetrics.init(servers, meterRegistry, clientTagProvider, 0);
+  }
+
+  @Nested
+  class MarkRequestProcessed {
+
+    @Test
+    public void happyPath() {
```

Comment:
can't we create more descriptive names for those tests that `happyPath`? and/or possibly use `@DisplayName` for a more human-readable test case? How the maven surefire output looks like when using N nested classes with N test suites where each is named `happyPath`?
```
 @DisplayName("Should calculate requests processed")
  public void shouldCalculateRequestsProcessed() {
```

Reply:
The `happyPath` name is to denote that we are testing a very simple straightforward path in the target method. It's usually there to test expected behavior. Since you know what's being tested from the nested class, it often does not make sence to now name it differently.. 

DisplayName is nice, but it's pure luxury to have and maintain it, as everything should be figured out from the test method name. If you take my tests names for example, you have smth like:

```
ClientMetricsTest.UpdateConnectedClients#happyPath
ClientMetricsTest.UpdateConnectedClients#noConnections
ClientMetricsTest.UpdateConnectedClients#removedConnections
```

* From `ClientMetricsTest` you know we are testing `ClientMetrics` class. 
* From `UpdateConnectedClients` you know we are testing `updateConnectedClients()` method
* From `#name` you know what specific situation are we testing in the method

Surefire reports looks good, more structured anyway that the flat test and quicker to realize what and where got wrong. But might I ask how often do you look into the surefire reports? When was the last time you looked?

Reply:
Sorry, by surefire I meant the standard `mvn` output when running test via `mvn test` command line.

Reply:
Same as for the other.

Reply:
> DisplayName is nice, but it's pure luxury to have and maintain it, as everything should be figured out from the test method name.

+1